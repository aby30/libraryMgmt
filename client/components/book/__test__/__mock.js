export const BookDetails =
{
    "bookId": 1,
    "availableQty": 1,
    "bookName": "BookName 1",
    "imageUrl": "https://res.cloudinary.com/abyy30/image/upload/v1608652378/bookImages/star.jpg",
    "isSelected": 0
};