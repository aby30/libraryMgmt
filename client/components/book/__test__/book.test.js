import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Book from '../index';
import { BookDetails } from './__mock.js';

afterEach(cleanup);

describe('Book card', () => {
    const addToCart = jest.fn();
    const removeFromCart = jest.fn();

    it('renders', () => {
        const { asFragment } = render(<Book bookDetails={BookDetails} addToCart={addToCart} removeFromCart={removeFromCart} />);
        expect(asFragment()).toMatchSnapshot();
    })

})