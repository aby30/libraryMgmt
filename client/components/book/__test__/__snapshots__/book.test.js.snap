// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book card renders 1`] = `
<DocumentFragment>
  <div
    class="book"
  >
    <div
      class="book__inner"
    >
      <img
        src="https://res.cloudinary.com/abyy30/image/upload/v1608652378/bookImages/star.jpg"
      />
      <div
        class="book__actionWrap"
      >
        <button
          class="button primary medium"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
